src_new/controllers/uas.js

```diff
@@ -0,0 +1,441 @@
+/* end-test-block */
+
+function executeDisplay(timeout, divIds, callback) {
+    if (UTIL.getExternalBidderStatus(divIds) && bidManager.getAllPartnersBidStatuses(window.PWT.bidMap, divIds)) {
+        UTIL.resetExternalBidderStatus(divIds);
+        callback();
+    } else {
+        (timeout > 0) && window.setTimeout(function() {
+          refThis.executeDisplay(timeout - 10, divIds, callback);
+        }, 10);
+    }
+}
+
+/* start-test-block */
+exports.executeDisplay = executeDisplay;
+/* end-test-block */
+
+function displayFunctionStatusHandler(oldStatus, theObject, originalFunction, arg) {
+    switch (oldStatus) {
+        // display method was called for this slot
+        /* istanbul ignore next */
+        case CONSTANTS.SLOT_STATUS.CREATED:
+        // dm flow is already intiated for this slot
+        // just intitate the CONFIG.getTimeout() now
+        // eslint-disable-line no-fallthrough
+        /* istanbul ignore next */
+        case CONSTANTS.SLOT_STATUS.PARTNERS_CALLED:
+            var divIds = Object.keys(refThis.slotsMap);
+
+            if (typeof window.OWT.externalBidderStatuses[arg[0]] === "object" && window.OWT.externalBidderStatuses[arg[0]]) {
+               refThis.executeDisplay(CONFIG.getTimeout(), divIds, function() {
+                   UTIL.forEachOnObject(refThis.slotsMap, function(key, slot) {
+                       refThis.findWinningBidIfRequired_Display(key, slot);
+                   });
+                   refThis.processDisplayCalledSlot(theObject, originalFunction, arg);
+                });
+            }
+
+            setTimeout(function() {
+              UTIL.log("PostTimeout.. back in display function");
+              UTIL.forEachOnObject(refThis.slotsMap, function(key, slot) {
+                  refThis.findWinningBidIfRequired_Display(key, slot);
+              });
+              refThis.processDisplayCalledSlot(theObject, originalFunction, arg);
+            }, CONFIG.getTimeout());
+
+            break;
+            // call the original function now
+        case CONSTANTS.SLOT_STATUS.TARGETING_ADDED:
+            refThis.updateStatusAndCallOriginalFunction_Display(
+                "As DM processing is already done, Calling original display function with arguments",
+                theObject,
+                originalFunction,
+                arg
+            );
+            break;
+
+        case CONSTANTS.SLOT_STATUS.DISPLAYED:
+            refThis.updateStatusAndCallOriginalFunction_Display(
+                "As slot is already displayed, Calling original display function with arguments",
+                theObject,
+                originalFunction,
+                arg
+            );
+            break;
+    }
+}
+
+/* start-test-block */
+exports.displayFunctionStatusHandler = displayFunctionStatusHandler;
+/* end-test-block */
+
+function initiateDisplay(win) {
+  win.Phoenix.registerPreDisplayHandler(function(originalDisplay, divId){
```

Comment:
We had discussed that eligible slots will be passed on to registered function by phoenix.js along with current arguments of phoenix.display function and a taskHandler object, it is not implemented

why originalDisplay is required

Reply:
Here originalDisplay is not actual originalDislay method its doneTask() from TaskManager will change name here.